### PR TITLE
Encode Strings with default external encoding

### PR DESCRIPTION
**Pull request by krissi**
This PR encodes strings from and to lua from/to the default external encoding

### CODE REVIEW

File: ext/rlua.c [hunk 1]
```diff
@@ -98,10 +106,17 @@ static void rlua_push_var(lua_State *state, VALUE value)
...
 
-    case T_STRING:
-      lua_pushlstring(state, RSTRING_PTR(value), RSTRING_LEN(value));
-      break;
+    case T_STRING: {
+      const char* string;
+#ifdef HAVE_RUBY_ENCODING_H
+      string = rb_str_export_to_enc(value, rb_default_external_encoding());
+#else
+      string = value
```
Comment (whitequark) [inline ext/rlua.c]: `;`. Actually, when is `HAVE_RUBY_ENCODING_H` not set? 1.8? We can ignore that.

**Comment by krissi** [comment]
yea, thats true. I took it from the psych source ;) Removed it

**Comment by krissi** [comment]
Could you release a new gem in the next days?

**Comment by whitequark** [comment]
1.1 released

**Pull request merged**